src/ACF/Sub_Field.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Tribe\Libs\ACF;
+
+class Sub_Field extends Field {
```

Comment:
I'm not really comfortable with this being a class. Consider a Repeater, a Layout, and a Group. Are those _kinds_ of Subfields, or do they _have_ subfields?

As a trait, it might look like:

```php
trait With_Sub_Fields {
	/** @var Field[] */
	protected $fields = [ ];

	/**
	 * @param Field $field
	 *
	 * @return static
	 */
	public function add_field( Field $field ) {
		$this->fields[] = $field;

		return $this;
	}

	public function get_sub_field_attributes(): array {
		return array_map( static function ( Field $f ) {
			return $f->get_attributes();
		}, $this->fields );
	}
}
```

And then usage like:

```php
class Repeater extends Field implements ACF_Aggregate {
	use With_Sub_Fields;

	public function __construct( $key, $attributes = [] ) {
		parent::__construct( $key, $attributes );
		$this->attributes[ 'type' ] = 'repeater';
	}

	public function get_attributes() {
		$attributes                 = parent::get_attributes();
		$attributes[ 'sub_fields' ] = $this->get_sub_field_attributes();
		return $attributes;
	}
}
```

I know this is all debatable; what are your thoughts?

Reply:
I actually had this exact thing and went the class route because 🤷  should be easy to get back. Thanks for the feedback.